Compliance: check-suite-prefs-unreadable

## Compliance Finding

**Category:** `settings`
**Severity:** `warning`
**Check:** `check-suite-prefs-unreadable`

## Detail

Could not read check-suite preferences — verify GH_TOKEN has repo scope and the repo is accessible. Check-suite auto-trigger compliance was not evaluated.

## Standard Reference

[standards/github-settings.md](https://github.com/petry-projects/.github/blob/main/standards/github-settings.md)

## Remediation

Run `scripts/apply-repo-settings.sh .github-private` with a token that has admin access to the repository (requires a classic PAT with `repo` scope or equivalent):

```bash
GH_TOKEN=<admin-pat> bash scripts/apply-repo-settings.sh .github-private
```

This script applies all standard settings defined in `standards/github-settings.md` in one pass.
For a dry run to preview changes without applying: `DRY_RUN=true GH_TOKEN=<admin-pat> bash scripts/apply-repo-settings.sh .github-private`

---
*This issue was automatically created by the [weekly compliance audit](https://github.com/petry-projects/.github/blob/main/.github/workflows/compliance-audit.yml).*

## Dev-Lead: Implementation Complete

### Plan Execution
- [x] Replaced the generic prefs-fetch failure warning in `rs_apply_repo` with an actionable diagnostic
- [x] Added a bats test asserting the diagnostic names `repo` scope and repo accessibility (and still returns 1)

### Context / Scope note
The finding's root cause is operational: the weekly compliance audit (in `petry-projects/.github`) could not read this repo's check-suite preferences because its token lacked `repo` scope / repo access. The documented remediation — `GH_TOKEN=<admin-pat> bash scripts/apply-repo-settings.sh .github-private` — requires an admin PAT and modifies shared repo settings, so it is out of scope for the dev-lead agent to execute (and is not a source change). **An admin still needs to run that remediation to clear the underlying warning.**

What *is* fixable in-repo: `rs_apply_repo` hits the exact same unreadable-prefs wall when an operator runs the remediation with an under-scoped token, but it emitted a generic `failed to get check-suite preferences` message that gave no next step. The diagnostic now mirrors the finding's guidance so the failure is self-explanatory.

### Test Results
```
$ bats tests/test_apply_repo_settings.bats
...
ok 16 rs_apply_repo live: returns 1 when GET check-suite preferences fails
ok 17 rs_apply_repo live: unreadable preferences warning names repo scope and accessibility
...
ok 32 live mode makes one PATCH call per required setting

32 tests, 0 failures

$ bash .dev-lead/scripts/dev-lead-lint.sh
  [lint] shellcheck on 67 file(s)...
  [lint] validating 4 agent profile(s)...
  [lint] all checks passed
```

### Files Changed
- `scripts/apply-repo-settings.sh`: actionable warning when check-suite preferences are unreadable — names the `repo` scope (classic PAT) requirement and repo accessibility. Behavior unchanged (still returns 1).
- `tests/test_apply_repo_settings.bats`: new test covering the improved diagnostic.

### Notes
- This is a diagnostic/UX hardening change for the script's failure path; it does not and cannot apply the repo settings or clear the compliance warning on its own. Clearing the warning requires an admin to run the remediation command above. Recommend keeping this issue open (or reassigning to an admin) until that operational step is performed.
- shellcheck on the modified file reports only a pre-existing `SC1091` info (cannot follow the sourced `lib/push-protection.sh`), which the warning-level repo lint ignores.

--- a/scripts/apply-repo-settings.sh
+++ b/scripts/apply-repo-settings.sh
@@ -64,7 +64,7 @@ rs_apply_repo() {
 
   local prefs
   if ! prefs=$(gh api "repos/${repo}/check-suites/preferences"); then
-    echo "[warn] ${repo}: failed to get check-suite preferences" >&2
+    echo "[warn] ${repo}: could not read check-suite preferences — verify GH_TOKEN has 'repo' scope (classic PAT) and the repo is accessible" >&2
     return 1
   fi
 

--- a/tests/test_apply_repo_settings.bats
+++ b/tests/test_apply_repo_settings.bats
@@ -179,6 +179,21 @@ teardown() {
   [ "$status" -eq 1 ]
 }
 
+@test "rs_apply_repo live: unreadable preferences warning names repo scope and accessibility" {
+  gh() {
+    if [[ "$*" == *"check-suites/preferences"* ]]; then
+      echo "[error] HTTP 404" >&2
+      return 1
+    fi
+  }
+  export -f gh
+  run rs_apply_repo "test-org/myrepo"
+  [ "$status" -eq 1 ]
+  [[ "$output" == *"repo"* ]]
+  [[ "$output" == *"scope"* ]]
+  [[ "$output" == *"accessible"* ]]
+}
+
 # ---------------------------------------------------------------------------
 # rs_apply_repo — live mode, already compliant
 # ---------------------------------------------------------------------------

--- a/tests/test_apply_repo_settings.bats
+++ b/tests/test_apply_repo_settings.bats
@@ -189,8 +189,7 @@ teardown() {
   export -f gh
   run rs_apply_repo "test-org/myrepo"
   [ "$status" -eq 1 ]
-  [[ "$output" == *"repo"* ]]
-  [[ "$output" == *"scope"* ]]
+  [[ "$output" == *"'repo' scope"* ]]
   [[ "$output" == *"accessible"* ]]
 }
 

